src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -145,200 +88,55 @@ void CCrossValidation::set_num_runs(int32_t num_runs)
 	m_num_runs = num_runs;
 }
 
-float64_t CCrossValidation::evaluate_one_run(
-    int64_t index, CrossValidationStorage* storage)
+float64_t CCrossValidation::evaluate_one_run(int64_t index) const
 {
 	SG_DEBUG("entering %s::evaluate_one_run()\n", get_name())
 	index_t num_subsets = m_splitting_strategy->get_num_subsets();
 
 	SG_DEBUG("building index sets for %d-fold cross-validation\n", num_subsets)
-
-	/* build index sets */
 	m_splitting_strategy->build_subsets();
 
-	/* results array */
 	SGVector<float64_t> results(num_subsets);
 
-	/* different behavior whether data is locked or not */
-	if (m_machine->is_data_locked())
+	#pragma omp parallel for shared(results)
+	for (auto i = 0; i<num_subsets; ++i)
 	{
-		m_machine->set_store_model_features(true);
-		SG_DEBUG("starting locked evaluation\n", get_name())
-		/* do actual cross-validation */
-		for (auto i : SG_PROGRESS(range(num_subsets)))
-		{
-			COMPUTATION_CONTROLLERS
-
-			/* evtl. update xvalidation output class */
-			CrossValidationFoldStorage* fold = new CrossValidationFoldStorage();
-			SG_REF(fold)
-			fold->set_run_index(index);
-			fold->set_fold_index(i);
-
-			/* index subset for training, will be freed below */
-			SGVector<index_t> inverse_subset_indices =
-			    m_splitting_strategy->generate_subset_inverse(i);
-
-			/* train machine on training features */
-			m_machine->train_locked(inverse_subset_indices);
-
-			/* feature subset for testing */
-			SGVector<index_t> subset_indices =
-			    m_splitting_strategy->generate_subset_indices(i);
-
-			/* evtl. update xvalidation output class */
-			fold->set_train_indices(inverse_subset_indices);
-			auto fold_machine = (CMachine*)m_machine->clone();
-			fold->set_trained_machine(fold_machine);
-			SG_UNREF(fold_machine)
-
-			/* produce output for desired indices */
-			CLabels* result_labels = m_machine->apply_locked(subset_indices);
```

Comment:
I removed the whole locked story....should be done properly or not at all ...this was just copy pasted code